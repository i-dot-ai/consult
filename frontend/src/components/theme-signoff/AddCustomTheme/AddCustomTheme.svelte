<script lang="ts">
  import Accordion from "../../Accordion/Accordion.svelte";
  import Panel from "../../dashboard/Panel/Panel.svelte";
  import Button from "../../inputs/Button/Button.svelte";
  import Textarea from "../../inputs/Textarea/Textarea.svelte";
  import TextInput from "../../inputs/TextInput/TextInput.svelte";
  import MaterialIcon from "../../MaterialIcon.svelte";
  import EditSquare from "../../svg/material/EditSquare.svelte";
  import Plus from "../../svg/material/Plus.svelte";
  import Tip from "../../svg/material/Tip.svelte";

  export interface Props {
    handleConfirm: (title: string, description: string) => void;
    handleCancel: () => void;
    variant?: "add" | "edit";
  }

  let {
    handleConfirm = () => {},
    handleCancel = () => {},
    variant = "add",
  }: Props = $props();

  let title = $state("");
  let description = $state("");

  const isThemeValid = () => {
    return Boolean(title && description);
  };
</script>

<Panel variant="primary" bg={true} border={true}>
  <div class="flex items-center gap-2">
    {#if variant === "add"}
      <MaterialIcon color="fill-primary">
        <Plus />
      </MaterialIcon>
    {:else}
      <MaterialIcon color="fill-primary">
        <EditSquare />
      </MaterialIcon>
    {/if}

    <h2 class="text-primary">
      {variant === "add" ? "Add Custom Theme" : "Edit Theme"}
    </h2>
  </div>

  <div class="mt-4 text-sm">
    <Accordion>
      {#snippet title()}
        <div class="flex gap-2 items-center">
          <MaterialIcon color="fill-neutral-500">
            <Tip />
          </MaterialIcon>
          Creating Effective Themes
        </div>
      {/snippet}

      {#snippet content()}
        <div class="text-xs">
          <h4>Good theme titles:</h4>
          <ul class="list-disc ml-4">
            <li>
              Are specific and descriptive (e.g., "Data Privacy Concerns")
            </li>
            <li>Use clear, jargon-free language</li>
            <li>Focus on the main topic or concern</li>
          </ul>

          <h4 class="mt-2">Good theme descriptions:</h4>
          <ul class="list-disc ml-4">
            <li>Explain what types of responses belong in this theme</li>
            <li>Include key concepts or phrases to look for</li>
            <li>Help categorise responses consistently</li>
          </ul>
        </div>
      {/snippet}
    </Accordion>
  </div>

  <div class="mt-4 text-sm">
    <TextInput
      id="title-input"
      label="Theme Title"
      placeholder="e.g., Data Privacy Concerns"
      value={title}
      setValue={(newTitle) => (title = newTitle)}
    />
  </div>

  <div class="mt-4 text-sm">
    <Textarea
      rows={3}
      id="description-input"
      label="Theme Description"
      placeholder="Describe what responses should be categorised under this theme..."
      value={description}
      setValue={(newVal) => (description = newVal)}
    />
  </div>

  <div class="mt-4">
    <Button
      variant="primary"
      size="sm"
      disabled={!isThemeValid()}
      handleClick={() => {
        handleConfirm( title, description );
      }}
    >
      Add Theme
    </Button>

    <Button size="sm" handleClick={() => handleCancel()}>Cancel</Button>
  </div>
</Panel>
