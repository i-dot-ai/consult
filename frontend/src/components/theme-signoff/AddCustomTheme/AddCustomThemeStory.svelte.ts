import AddCustomTheme from "./AddCustomTheme.svelte";

let variant = $state("add");
const handleConfirm = () => alert("Confirm event triggered");
const handleCancel = () => alert("Cancel event triggered");


export default {
  name: "AddCustomTheme",
  component: AddCustomTheme,
  category: "Theme Signoff",
  props: [
    {
      name: "variant",
      value: variant,
      type: "select",
      options: [
        { value: "add", label: "Add" },
        { value: "edit", label: "Edit" },
      ],
    },
    {
      name: "handleConfirm",
      value: handleConfirm,
      type: "func",
      schema: "(title: string, description: string) => void",
    },
    {
      name: "handleCancel",
      value: handleCancel,
      type: "func",
      schema: "() => void",
    },
  ],
  stories: [
    {
      name: "Add Variant",
      props: {
        variant: "add",
        handleConfirm: handleConfirm,
        handleCancel: handleCancel,
      },
    },
    {
      name: "Edit Variant",
      props: {
        variant: "edit",
        handleConfirm: handleConfirm,
        handleCancel: handleCancel,
      },
    },
  ],
};
