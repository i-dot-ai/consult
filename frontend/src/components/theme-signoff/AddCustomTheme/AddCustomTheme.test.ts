import { afterEach, beforeEach, describe, expect, it } from "vitest";
import { render, cleanup } from "@testing-library/svelte";

import AddCustomTheme, { type Props } from "./AddCustomTheme.svelte";

let testData: Props;

describe("AddCustomTheme", () => {
  beforeEach(() => {
    testData = {
      variant: "add",
      handleConfirm: () => {},
      handleCancel: () => {},
    };
  });

  afterEach(() => cleanup());

  it("should render add variant", async () => {
    const { container, getByText } = render(AddCustomTheme, {
      ...testData,
    });

    expect(getByText("Add Custom Theme"));

    expect(container).toMatchSnapshot();
  });

  it("should render add variant", async () => {
    const { container, getByText } = render(AddCustomTheme, {
      ...testData,
      variant: "edit",
    });

    expect(getByText("Edit Theme"));
  });
});
