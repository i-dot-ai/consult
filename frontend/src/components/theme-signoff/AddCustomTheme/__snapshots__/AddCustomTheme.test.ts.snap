// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddCustomTheme > should render add variant 1`] = `
<div>
  <div
    class="h-full my-4 p-4 rounded-2xl border border-pink-200 bg-pink-50"
    data-testid="panel-component"
  >
    <div
      class="flex items-center gap-2"
    >
      <svg
        class="fill-primary"
        height="1rem"
        role="img"
        viewBox="0 -960 960 960"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M440-440H200v-80h240v-240h80v240h240v80H520v240h-80v-240Z"
          xmlns="http://www.w3.org/2000/svg"
        />
        <!---->
      </svg>
      <!---->
       
      <h2
        class="text-primary"
      >
        Add Custom Theme
      </h2>
    </div>
     
    <div
      class="mt-4 text-sm"
    >
      <!---->
      <!---->
      <button
        aria-pressed="false"
        class="w-full text-md cursor-pointer rounded-md py-1 px-2 border bg-white border-gray-300 transition-colors duration-300 self-start inline-flex gap-1 items-center hover:bg-gray-100 svelte-1rjb6fa"
        data-variant="default"
        role="button"
        tabindex="0"
      >
        <!---->
        <div
          class="flex justify-between items-center gap-2 w-full"
        >
          <div
            class="flex gap-2 items-center"
          >
            <svg
              class="fill-neutral-500"
              height="1rem"
              role="img"
              viewBox="0 -960 960 960"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M440-280h80v-240h-80v240Zm40-320q17 0 28.5-11.5T520-640q0-17-11.5-28.5T480-680q-17 0-28.5 11.5T440-640q0 17 11.5 28.5T480-600Zm0 520q-83 0-156-31.5T197-197q-54-54-85.5-127T80-480q0-83 31.5-156T197-763q54-54 127-85.5T480-880q83 0 156 31.5T763-763q54 54 85.5 127T880-480q0 83-31.5 156T763-197q-54 54-127 85.5T480-80Zm0-80q134 0 227-93t93-227q0-134-93-227t-227-93q-134 0-227 93t-93 227q0 134 93 227t227 93Zm0-320Z"
                xmlns="http://www.w3.org/2000/svg"
              />
              <!---->
            </svg>
            <!---->
             Creating Effective Themes
          </div>
          <!---->
           
          <div
            class="transition-transform"
          >
            <svg
              class="fill-neutral-500"
              height="1rem"
              role="img"
              viewBox="0 -960 960 960"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504-480 320-664l56-56 240 240-240 240-56-56 184-184Z"
                xmlns="http://www.w3.org/2000/svg"
              />
              <!---->
            </svg>
            <!---->
          </div>
        </div>
        
        
      </button>
      
      <!---->
       
      <!---->
      <!---->
    </div>
     
    <div
      class="mt-4 text-sm"
    >
      <div
        class="relative"
      >
        <label
          for="title-input"
        >
          Theme Title
        </label>
         
        <input
          class="w-full mt-1 p-1 border border-gray-300 rounded-sm focus:outline-2 focus:outline-yellow-300 svelte-1bet92c"
          id="title-input"
          placeholder="e.g., Data Privacy Concerns"
          type="text"
        />
         
        <!---->
      </div>
      <!---->
    </div>
     
    <div
      class="mt-4 text-sm"
    >
      <div>
        <label
          for="description-input"
        >
          Theme Description
        </label>
         
        <textarea
          class="block w-full mt-1 p-1 border border-gray-300 rounded-sm focus:outline-2 focus:outline-yellow-300"
          id="description-input"
          placeholder="Describe what responses should be categorised under this theme..."
          rows="3"
          style="min-height: 3lh;"
        />
      </div>
      <!---->
    </div>
     
    <div
      class="mt-4"
    >
      <!---->
      <button
        aria-pressed="false"
        class="text-sm cursor-pointer rounded-md py-1 px-2 border bg-primary text-white border-gray-300 transition-colors duration-300 self-start inline-flex gap-1 items-center hover:bg-gray-100 hover:text-primary hover:border-primary disabled disabled:bg-gray-200 disabled:text-gray-400 disabled:cursor-not-allowed disabled:hover:bg-gray-200 disabled:hover:border-gray-300 disabled:hover:text-gray-400 svelte-1rjb6fa"
        data-variant="primary"
        disabled=""
        role="button"
        tabindex="0"
      >
        <!---->
        Add Theme
        
        
      </button>
      
      <!---->
       
      <!---->
      <button
        aria-pressed="false"
        class="text-sm cursor-pointer rounded-md py-1 px-2 border bg-white border-gray-300 transition-colors duration-300 self-start inline-flex gap-1 items-center hover:bg-gray-100 svelte-1rjb6fa"
        data-variant="default"
        role="button"
        tabindex="0"
      >
        <!---->
        Cancel
        
        
      </button>
      
      <!---->
    </div>
    <!---->
  </div>
  
</div>
`;
